src/disk_io_thread.cpp

```diff
@@ -361,8 +362,9 @@ namespace libtorrent
 		}
 
 		// piece range
-		int range_start = int((p->piece / cont_pieces) * cont_pieces);
-		int range_end = (std::min)(range_start + cont_pieces, p->storage->files()->num_pieces());
+		piece_index_t const range_start((static_cast<int>(p->piece) / cont_pieces) * cont_pieces);
+		piece_index_t const range_end(std::min(static_cast<int>(range_start)
+			+ cont_pieces, p->storage->files()->num_pieces()));
```

Comment:
I use ``end_piece()`` as the one-past-end piece index for loops. It probably makes sense to use here too, but this is one of few places where I do actual arithmetic on piece indices, so it's a bit of an exception since it needs to be converted to ``int`` anyway.